vllm/model_executor/layers/logits_processor.py

```diff
@@ -84,21 +84,29 @@ def _apply_logits_processors(
     logits: torch.Tensor,
     sampling_metadata: SamplingMetadata,
 ) -> torch.Tensor:
-    logits_row_idx = 0
+    logits_row_idx = -1
     found_logits_processors = False
     for seq_ids, sampling_params in sampling_metadata.seq_groups:
         logits_processors = sampling_params.logits_processors
         if logits_processors:
             found_logits_processors = True
             for seq_id in seq_ids:
+                # handle prompt_logprobs by skipping rows in logits added for
+                # the prompt tokens (prompt logprobs are not processed)
+                if (seq_id < sampling_metadata.num_prompts
```

Comment:
`seq_id`s are allocated using a global counter, so you'll have to use `enumerate` as is done in https://github.com/vllm-project/vllm/pull/3876.

Reply:
I'm not totally sure what happens when `seq_ids` has more than one value. I hope the new fix is on point, especially since it looks like `logits_row_idx` might get added up multiple times for each `seq_id`.

Reply:
> it looks like `logits_row_idx` might get added up multiple times for each `seq_id`.

There should only be one `seq_id` in the prompt case. However I think these rows need to be skipped whether or not the request has logits processors (see my comment below).